### ISSUE TITLE
Tracking issue for future-incompatibility lint `cenum_impl_drop_cast`

### ISSUE THREAD

**Issue by oddg** [rust-lang/rust#73333]
This is the **summary issue** for the `CENUM_IMPL_DROP_CAST` future-incompatibility lint and other related errors. The goal of this page is describe why this change was made and how you can fix code that is affected by it. It also provides a place to ask questions or register a complaint if you feel the change should not be made. For more information on the policy around future-compatibility warnings, see our [breaking change policy guidelines][guidelines].

[guidelines]: https://github.com/rust-lang/rfcs/blob/master/text/1122-language-semver.md

#### What is the warning for?

The warning is issue when a C-like enum implementing the `Drop` trait is cast. For example:

```rust
enum E {
    A = 0,
}

impl Drop for E {
    fn drop(&mut self) {
        println!("Drop");
    }
}

fn main() {
    let e = E::A;
    let i = e as u32;
            ^^^^^^^^ here is the cast
}
```

This cast will be rejected to enforce uniformity between all moving operations. See https://github.com/rust-lang/rust/issues/35941 for more details.

#### When will this warning become a hard error?

At the beginning of each 6-week release cycle, the Rust compiler team will review the set of outstanding future compatibility warnings and nominate some of them for **Final Comment Period**. Toward the end of the cycle, we will review any comments and make a final determination whether to convert the warning into a hard error or remove it entirely.

**Comment by RalfJung** [comment]
In https://github.com/rust-lang/rust/pull/97652 I am proposing to make this deny-by-default and make it show up in future breakage reports.
